### PR TITLE
Smart configure options

### PR DESCRIPTION
**Pull request by Womblue**
This PR enables the use of a new config option in `flight-profile-types` to prefill each configuration question a `default_smart` command, specified in the type's metadata file for each question. Any command placed here will be run, and its output will be given as the prefilled entry if the result is non-empty. For example:

```
  - id: access_host
    env: ACCESS_HOST
    text: 'IP or FQDN for Web Access:'
    default_smart: "curl -s ifconfig.me"
    validation:
      type: string
```
would give the node's public IP as a default value.

The new priority order for these prefilled values is as follows:
* The existing value, if it has already been set previously
* The result of `default_smart`, if it exists and the command produces a result
* The `default` entry string given for that question

If none of the above are given, no prefilled answer is provided.

### CODE REVIEW

File: lib/profile/commands/configure.rb [hunk 1]
```diff
@@ -31,14 +31,31 @@ def display_details
...
           type.questions.each do |question|
             key(question.id).ask(question.text) do |q|
-              if type.fetch_answer(question.id)
-                q.default type.fetch_answer(question.id)
-              elsif question.default
-                q.default question.default
+
+              prefill = type.fetch_answer(question.id)
+              if question.default_smart
+                process = Flight::Subprocess::Local.new(
```
Comment (VoxSecundus) [inline lib/profile/commands/configure.rb:41]: Good idea using `Flight::Subprocess` here.

File: lib/profile/commands/configure.rb [hunk 2]
```diff
@@ -31,14 +31,31 @@ def display_details
...
+                  env: {},
+                  logger: smart_log,
+                  timeout: 5,
+                )
+                result = process.run(question.default_smart, nil)
+                output = result.stdout.chomp
+                if !result.success?
+                  smart_log.debug("Command '#{question.default_smart}' failed to run: #{result.stderr}")
```
Comment (VoxSecundus) [inline lib/profile/commands/configure.rb:49]: I don't think that we need a logger here. I think that if the command doesn't return successfully, we should just fall back to the next default option.

File: lib/profile/commands/configure.rb [hunk 3]
```diff
@@ -31,14 +31,31 @@ def display_details
...
         @answers = prompt.collect do
           type.questions.each do |question|
             key(question.id).ask(question.text) do |q|
-              if type.fetch_answer(question.id)
-                q.default type.fetch_answer(question.id)
-              elsif question.default
-                q.default question.default
+
+              prefill = type.fetch_answer(question.id)
+              if question.default_smart
```
Comment (VoxSecundus) [inline lib/profile/commands/configure.rb:40]: If the question already has an answer, and we know we'll be using that answer in the `default` field, we don't need to run the `default_smart` command.

**Review by VoxSecundus — CHANGES_REQUESTED** [review]
Nice work; couple of efficiency changes to make.

File: lib/profile/commands/configure.rb (hunk 3 above)
Comment (Womblue) [reply lib/profile/commands/configure.rb]: Addressed in newest version

**Review by VoxSecundus — APPROVED** [review]
LGTM; thanks for your work on this

**Pull request merged**